backend/app/server/app.py

```diff
@@ -48,22 +49,47 @@ async def get_shop(shop_id: str):
     shop = await Shop.get(shop_id)
     return shop
 
-
-@app.get("/shops/by_distance", tags=["Shops"])
+@app.post("/shops/by_distance", tags=["Shops"])
 async def get_shops_by_dist(query: ShopsByDistance):
     r = query.r
     lat = query.lat
     long = query.long
-    shops = await Shop.filter_by_distance(
-        localization=(lat, long), radius=r).to_list()
-    return shops
-
 
-@app.get("/shops/by_number", tags=["Shops"])
-async def get_n_nearest_shops(query: ShopsByDistance):
+    geo_near_stage = {
+        "$geoNear": {
+            "near": {
+                "type": "Point",
+                "coordinates": [lat, long]
+            },
+            "spherical": True,
+            "distanceField": "distance",
+            "maxDistance": r
+        }
+    }
+
+    result = await Shop.get_motor_collection().aggregate([geo_near_stage]).to_list(None)
+
+    shops = [Shop(**doc) for doc in result]
+    return [shop.dict() for shop in shops]
```

Comment:
Nie musisz tutaj zmieniać na nowo na słownik. Jak zwrócisz shops to będzie działać